File: pset6/mario.py

def print_row(is_left, width, height):
    num_blanks = height - width
    if is_left:
        place_blanks(num_blanks)
        place_blocks(width)
    else:
        place_blocks(width)
        print()
        
def place_blocks(num_blocks):
    for i in range(num_blocks):
        print("#", end='')
        
def place_blanks(num_blanks):
    for i in range(num_blanks):
        print(" ", end='')
        
def get_height():
    while True:
        n = int(input("Enter a height for the pyramids: "))
        if n > 0 and n < 23:
            break
    return n

if __name__ == "__main__":
    height = get_height()
    width = 1
    for i in range(height):
        print_row(True, width, height)
        print(" ", end='')
        print_row(False, width, height)
        width += 1

File: pset1/mario.c
#include <stdio.h>
#include <cs50.h>

void placeBlocks(int numBlocks);
void placeBlanks(int numBlanks);
void printRow(bool isLeft, int width, int height);
int getHeight();

int main(void)
{
    int height = getHeight();
    int width = 1;
    for (int i = 0; i < height; i++)
    {
        printRow(true, width, height);
        printf("  ");
        printRow(false, width, height);
        width += 1;
    }
}

void printRow(bool isLeft, int width, int height)
{
    int numBlanks = height - width;
    if (isLeft)
    {
        placeBlanks(numBlanks);
        placeBlocks(width);
    }
    else
    {
        placeBlocks(width);
        printf("\n");
    }
}

void placeBlocks(int numBlocks)
{
    for (int i = 0; i < numBlocks; i++)
    {
        printf("#");
    }
}

void placeBlanks(int numBlanks)
{
    for (int i = 0; i < numBlanks; i++)
    {
        printf(" ");
    }
}

int getHeight()
{
    int n;
    do
    {
        printf("Enter a height for the pyramids: ");
        n = get_int();
    }
    while (n < 0 || n > 23);
    return n;
}